src/Core/Math/Ellipsoid.js

```diff
@@ -1,182 +1,162 @@
+     * @param {number} position.z
+     * @param {Coordinate} [target] coordinate to copy result
+     * @returns {Coordinate} an object describing the coordinates on the reference ellipsoid, angles are in degree
+     */
+    cartesianToCartographic(position, target = new Coordinates('EPSG:4326', 0, 0, 0)) {
+        // for details, see for example http://www.linz.govt.nz/data/geodetic-system/coordinate-conversion/geodetic-datum-conversions/equations-used-datum
+        // TODO the following is only valable for oblate ellipsoid of revolution. do we want to support triaxial ellipsoid?
+        const R = Math.sqrt(position.x * position.x + position.y * position.y + position.z * position.z);
+        const a = this.size.x; // x
+        const b = this.size.z; // z
+        const e = Math.abs((a * a - b * b) / (a * a));
+        const f = 1 - Math.sqrt(1 - e);
+        const rsqXY = Math.sqrt(position.x * position.x + position.y * position.y);
+
+        const theta = Math.atan2(position.y, position.x);
+        const nu = Math.atan(position.z / rsqXY * ((1 - f) + e * a / R));
+
+        const sinu = Math.sin(nu);
+        const cosu = Math.cos(nu);
+
+        const phi = Math.atan((position.z * (1 - f) + e * a * sinu * sinu * sinu) / ((1 - f) * (rsqXY - e * a * cosu * cosu * cosu)));
+
+        const h = (rsqXY * Math.cos(phi)) + position.z * Math.sin(phi) - a * Math.sqrt(1 - e * Math.sin(phi) * Math.sin(phi));
+
+        return target.set('EPSG:4326',
+            THREE.Math.radToDeg(theta),
+            THREE.Math.radToDeg(phi),
+            h);
+    }
 
-Ellipsoid.prototype.intersection = function intersection(ray) {
-    var EPSILON = 0.0001;
-    var O_C = ray.origin;
-    var dir = ray.direction;
-    // normalizeVector( dir );
+    cartographicToCartesianArray(coordCartoArray) {
+        var cartesianArray = [];
+        for (var i = 0; i < coordCartoArray.length; i++) {
+            cartesianArray.push(this.cartographicToCartesian(coordCartoArray[i]));
+        }
 
-    var a =
-        ((dir.x * dir.x) / (this.size.x * this.size.x)) + ((dir.y * dir.y) / (this.size.y * this.size.y)) + ((dir.z * dir.z) / (this.size.z * this.size.z));
+        return cartesianArray;
+    }
 
-    var b =
-        ((2 * O_C.x * dir.x) / (this.size.x * this.size.x)) + ((2 * O_C.y * dir.y) / (this.size.y * this.size.y)) + ((2 * O_C.z * dir.z) / (this.size.z * this.size.z));
-    var c =
-        ((O_C.x * O_C.x) / (this.size.x * this.size.x)) + ((O_C.y * O_C.y) / (this.size.y * this.size.y)) + ((O_C.z * O_C.z) / (this.size.z * this.size.z)) - 1;
+    intersection(ray) {
+        var EPSILON = 0.0001;
+        var O_C = ray.origin;
+        var dir = ray.direction;
+        // normalizeVector( dir );
 
-    var d = ((b * b) - (4 * a * c));
-    if (d < 0 || a === 0 || b === 0 || c === 0) { return false; }
+        var a =
+            ((dir.x * dir.x) / this._radiiSquared.x) + ((dir.y * dir.y) / this._radiiSquared.y) + ((dir.z * dir.z) / this._radiiSquared.z);
```

Comment:
sorry use this.`this._invRadiiSquared` instead of `this._radiiSquared`

Reply:
Done, and rebased.